### PR TITLE
updates to make clear root can be catalog or collection

### PR DESCRIPTION
**Pull request by cholmes**
**Related Issue(s):** #1082 

**Proposed Changes:**

1. Various little tweaks to make it clear that the 'root' can be a Catalog or Collection
2. Or small language tightening that came up as reading.

**PR Checklist:**

- [x] This PR is made against the dev branch (all proposed changes except releases should be against dev, not master).
- [x] This PR has **no** breaking changes.
- [x]  a CHANGELOG entry is not required.

### CODE REVIEW

File: overview.md [hunk 1]
```diff
@@ -91,14 +92,14 @@ points to [STAC Items](item-spec/README.md), or to other STAC catalogs. It provi
 linking structure that can be used recursively so that many Items can be included in 
 a single Catalog, organized however the implementor desires. 
 
-STAC makes no formal distinction between a "root" catalog and the "child" catalogs. A root catalog
-is simply the top-most catalog -- it has no parent. A nested catalog structure is useful (and
+STAC makes no formal distinction between a "root" Catalog and the "child" Catalogs. A root Catalog
+is simply the top-most Catalog -- it has no parent. A nested catalog structure is useful (and
```
Comment (m-mohr) [inline overview.md:96]:
```suggestion
is simply the top-most Catalog or Collection -- it has no parent. A nested catalog structure is useful (and
```

File: overview.md [hunk 2]
```diff
@@ -137,6 +138,9 @@ root catalog might be a sub-catalog of someone else's structure. The goal is for
 information and links they want to, while also encouraging a natural web of information to arise as Catalogs and Items are
 linked to across the web.
 
+*NOTE: The above examples all say Catalog, but those can all be Collections as well, as it has all the fields necessary to
```
Comment (m-mohr) [inline overview.md:141]: Maybe put at the top where it starts to get important to say this? May reduce confusion...

**Comment by m-mohr** [comment]
There are a lot more things to change, for example, numerous mentions of Catalogs in Best Practices, which should always mention both Catalogs and Collections.

The item spec also has a mention of "root Catalog" in the description of the ID field.

**Review by m-mohr — CHANGES_REQUESTED** [review]
There are more things to change, for example, numerous mentions of Catalogs in Best Practices, which should always mention both Catalogs and Collections.

The item spec also has a mention of "root Catalog" in the description of the ID field.

**Review by jbants — APPROVED** [review]

**Review by matthewhanson — APPROVED** [review]

**Review by jisantuc — APPROVED** [review]

**Comment by m-mohr** [comment]
Some more parts, which seem to apply to both Catalog and Collection:

### best-practices (I haven't read it completely):

> Sub-Catalogs should be stored in subdirectories of their parent

> Items should be stored in subdirectories of their parent Catalog. 

> Limit the number of Items in a Catalog or sub-Catalog, grouping / partitioning as relevant to the dataset.

> providing multiple 'sub-catalogs'
> from the root Catalog that enable different paths to browse

> This is a self-contained catalog as described above, except it includes an absolute `self` link at
> the root catalog

### catalog-spec:

> Every STAC Collection is 
> also a valid STAC Catalog.

Not true in validation due to the type field.

> **Note:** A link to at least one `item` or `child` Catalog is **REQUIRED**.

One child link is required, can be Catalog or Collection.

### collection-spec:

> Summaries should summarize all values in every Item underneath the collection, including in any nested sub-Catalogs.

> | parent  | URL to the parent STAC entity (Catalog or Collection). Non-root Collections should include a link to their parent. |
> | child   | URL to a child STAC Catalog or Collection. |

Why is it sometimes "STAC entity (Catalog or Collection)" and sometimes "STAC Catalog or Collection"?

**Comment by cholmes** [comment]
Weird, I had a couple of those on my local copy but didn't get them committed. Will do the rest and commit.

**Comment by cholmes** [comment]
I think I got them all.

**Review by m-mohr — APPROVED** [review]
Approving after a couple of additional changes, please check them @cholmes and then you can merge.

**Pull request merged**